File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 7/14/2026, 2:06:21 PM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        r = ""
        for i in range(len(strs[0])):
            ch = strs[0][i]
            for j in strs[1:]:
                if i >= len(j) or j[i] != ch:
                    return r  
            r += ch  
        return r

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.java
// Last updated: 7/14/2026, 2:06:15 PM
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String r="";
        for(int i=0;i<strs[0].length();i++){
            char c=strs[0].charAt(i);
            for(int j=1;j<strs.length;j++){
                if(i>=strs[j].length() || strs[j].charAt(i)!=c){
                    return r;
                }
            }r=r+c;
        }return r;
    }
}